### ISSUE TITLE
install.sh fails to find valid release in CI (GitHub API rate limiting)

### ISSUE THREAD

**Issue by kaovilai** [golangci/golangci-lint#6259]
### Welcome

- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

install.sh script

### Description of the problem

The install.sh script fails to find valid releases when run in CI environments (e.g., GitHub Actions during Docker builds), reporting "unable to find 'v2.5.0'" even though the release exists with all necessary assets.

**Reproduction:**
```bash
# In CI environment (GitHub Actions, Docker build)
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /tmp/test v2.5.0
```

**Error output:**
```
golangci/golangci-lint info checking GitHub for tag 'v2.5.0'
golangci/golangci-lint crit unable to find 'v2.5.0' - use 'latest' or see https://github.com/golangci/golangci-lint/releases for details
```

**However:**
- The release exists: https://github.com/golangci/golangci-lint/releases/tag/v2.5.0
- Has all necessary binaries (linux-amd64, darwin-arm64, etc.)
- Local installation works fine
- Installation in Docker locally works fine

### Root Cause

The install script's `github_release()` function makes unauthenticated GitHub API calls:
```bash
giturl="https://github.com/${owner_repo}/releases/${version}"
json=$(http_copy "$giturl" "Accept:application/json")
```

In CI environments with many builds, these unauthenticated API calls hit GitHub's rate limits, causing the script to fail even for valid releases.

### Proposed Solution

1. **Support GITHUB_TOKEN environment variable** for authenticated API requests
2. **Add retry logic** with exponential backoff for rate limit errors
3. **Document the go install alternative** as the recommended CI approach:
   ```bash
   go install github.com/golangci/golangci-lint/cmd/golangci-lint@v2.5.0
   ```

The `go install` method is more reliable in CI since it doesn't hit GitHub API rate limits and is the official Go way to install tools.

### Version of golangci-lint

v2.5.0 (and likely affects other versions)

### Go environment

```
Docker: golang:1.25-bookworm
GOOS=linux
GOARCH=amd64
```

### Additional context

This appears to be a CI-specific issue related to GitHub API rate limiting. The install script works fine in environments with fewer API requests.

> [!Note]
> Responses generated with Claude

**Comment by kaovilai** [comment]
> [!Note]
> Responses generated with Claude

Related downstream issue: https://github.com/vmware-tanzu/velero/issues/9450

This is affecting Velero's CI builds where the install script is called during Docker image builds.

**Comment by ldez** [comment]
Humanity's razor:

What an AI has created can be dismissed without explanation.

**Comment by kaovilai** [comment]
Humanity response: I will now have to recreate the issue, yay.

**Comment by ldez** [comment]
https://github.com/vmware-tanzu/velero/issues/9450#issuecomment-3643254246

**Comment by kaovilai** [comment]
kk won't touch this issue to bother you anymore here. but to correct one thing.. The info in this bug is correct
there can absolutely be rate limits with unauthenticated http requests (as you noted in our repo issue).

https://github.blog/changelog/2025-05-08-updated-rate-limits-for-unauthenticated-requests/

API calls with auth will have higher limits. It's easier to hit in ci environment the limit especially on shared github actions runner IP ironically.

**Comment by ldez** [comment]
There is no API calls inside the script, so there is no rate limit problem.

The script calls: 
- https://github.com/golangci/golangci-lint/releases/${version}
- https://github.com/golangci/golangci-lint/releases/download/${TAG}/${TARBALL}

Those two elements are **not** related to the GitHub API (`https://api.github.com`).
